### ISSUE TITLE
Question regarding HSTS "cookies"

### ISSUE THREAD

**Issue by brownie-hawkeye** [bleachbit/bleachbit#1677]
Does BleachBit delete the HSTS "cookies" when it cleans?

**Comment by XhmikosR** [comment]
There's no HSTS cookie...

**Comment by brownie-hawkeye** [comment]
> There's no HSTS cookie...

That's why I put HSTS cookie in quotes.

 A few questions :




What are HSTS Super Cookies?
https://security.stackexchange.com/questions/79518/what-are-hsts-super-cookies

    What are HSTS super cookie?
    How do they work?
    Why is possible to access them from multiple domains?
    Do I have to worry and if yes, how to mitigate their effect?




I do hope that the good folk at bleachbit are aware of the HSTS super cookies ...

**Comment by az0** [comment]
> Does BleachBit delete the HSTS "cookies" when it cleans?

See below. How do you feel about the tradeoffs?

## tl;dr 

Firefox: yes.

Chromium: not in BleachBit 6.0.2

## Trade offs

As with many cleaning options, there are tradeoffs with HSTS. While HSTS information can be used for fingerprinting, HSTS prevents MITM attacks via downgrade attacks, so there is a tradeoff. 

Visiting insecure sites is mitigated by

- Popular websites pre-load HSTS information is browsers (without fingerprinting/supercookies)
- Chrome 154 in October 2026 [will enable “Always Use Secure Connections”](https://blog.google/security/https-by-defau/)
- In Chrome, go to `chrome://settings/security` and **Always use secure connections**
- In Firefox, go to `about:preferences#connectionSecurity` and enable **HTTPS-Only Mode**

However, requiring secure connections doesn't replace the need for HSTS: HSTS is stronger.

## Chromium

In Chromium-browsers, the HSTS info is in `TransportSecurity`. BleachBit 6.0.2 does not delete it, but you can add it right now in **Preferences** - **Custom**. After listing the file there, enable the option **Custom** under the **System** cleaner.

Here is a test procedure to verify deleting the file removes dynamic HSTS info.

1. Visit https://wikipedia.org and https://hsts.badssl.com
2. Navigate to chrome://net-internals/#hsts
3. Under **Query HSTS/PKP domain** type their hostnames (without `https://`) 

You will see something like this

```
dynamic_sts_domain: hsts.badssl.com
dynamic_upgrade_mode: FORCE_HTTPS
dynamic_sts_include_subdomains: true
dynamic_sts_observed: (long number)
dynamic_sts_expiry: (long number)
```

4. Close browser
5. Delete `TransportSecurity` under the profile directory (usually `Default`)
6. Re-open browser but don't visit those sites
7. Repeat steps 2 and 3. For wikipedia, the static section (top) will have data, but then should show this

```
static_sts_domain: wikipedia.org
static_upgrade_mode: FORCE_HTTPS
static_sts_include_subdomains: true
static_sts_observed: 1785961062
static_pkp_domain:
static_pkp_include_subdomains:
static_pkp_observed:
static_spki_hashes:
dynamic_sts_domain:
dynamic_upgrade_mode: UNKNOWN
dynamic_sts_include_subdomains:
dynamic_sts_observed:
dynamic_sts_expiry:
```

8. Check `hsts.badssl.com`: it should be **not found **

## Firefox

For years, BleachBit cleaned the old `SiteSecurityServiceState.txt`, and recently, 5d18ca added support for the new `SiteSecurityServiceState.bin`. **Yes: both HSTS file names are supported fully in Firefox and Waterfox**, while Librewolf, Zen, and Palemoon don't have the `.bin` variation.

Firefox doesn't have a internal URL like Chrome to view HSTS information.

**Comment by az0** [comment]
LibreWolf 149.0-1 does not record dynamic HSTS info (in either .bin or .txt), so there is nothing to delete.

Palemoon 34.3.2.1 (64-bit) still makes `SiteSecurityServiceState.txt` (not .bin).

Zen 1.21.15b and Waterfox _do_ make "SiteSecurityServiceState.bin"

**Comment by az0** [comment]
[BleachBit 6.0.4 (stable)](https://www.bleachbit.org/news/bleachbit-604) is released with new actions to clean HSTS dynamic data for supercookies
